code/datums/emergency_calls/emergency_call.dm

```diff
@@ -39,23 +46,28 @@
 
 //Randomizes and chooses a call datum.
 /datum/game_mode/proc/get_random_call()
-	var/datum/emergency_call/chosen_call
-	var/list/valid_calls = list()
-
+	var/normalised_monitor_state = SSmonitor.current_points / XENOS_LOSING_THRESHOLD
+	var/list/calls_weighted = list()
 	for(var/datum/emergency_call/E in all_calls) //Loop through all potential candidates
-		if(E.probability < 1) //Those that are meant to be admin-only
+		if(E.base_probability <= 0)
 			continue
-
-		valid_calls.Add(E)
-
-		if(prob(E.probability))
-			chosen_call = E
-			break
-
-	if(!istype(chosen_call))
-		chosen_call = pick(valid_calls)
-
-	return chosen_call
+		calls_weighted[E] = E.get_actualised_weight(normalised_monitor_state)
+
+	return pickweight(calls_weighted)
+
+/** Return a new current_weight using the base probability, the allignement factor of the ERT and the monitor state
+ * monitor_state : the normalised state of the monitor. If it's equal to -1, monitor is barely in its MARINE_LOSING state. 
+ * A +2.5 value mean we are beyond the XENO_DELAYING state, aka marines have crushed the xenos
+ */
+/datum/emergency_call/proc/get_actualised_weight(monitor_state)
+	var/probability_direction = (monitor_state * allignement_factor)
+	var/actualised_weight = base_probability
+	if(probability_direction >= 0)
+		actualised_weight *= (1+probability_direction)
+	else
+		actualised_weight /= (1-probability_direction)
+	message_admins("[name] has a weight of [actualised_weight]")
+	return actualised_weight
```

Comment:
You can directly return here, debug message admins, actualized weight isnt needed

Reply:
yeah i know, i'm still testing things up